models/skipthoughts.py

```diff
@@ -0,0 +1,194 @@
+"""
+A simple model based on skipthoughts sentence embeddings.
+
+To set up:
+    * Execute the "Getting started" wgets in its README
+    * set config['skipthoughts_datadir'] to directory with downloaded files
+    * make skipthoughts.py from https://github.com/ryankiros/skip-thoughts/blob/master/skipthoughts.py 
+        available via import skipthoughts
+
+Inner working: First we compute skipthought embedding of both inputs; then we merge them (multiply & subtract), cancatenate, and compute result (1 MLP layer).
+"""
+
+from __future__ import print_function
+from __future__ import division
+
+
+from keras.models import Graph
+from keras.layers.core import Activation, Dense, Dropout
+from keras.regularizers import l2
+from keras.optimizers import Adam
+
+import pysts.embedding as emb
+import pysts.loader as loader
+import pysts.kerasts.blocks as B
+from pysts.kerasts.objectives import pearsonobj
+
+import numpy as np
+
+
+def config(c):
+    # XXX:
+    c['skipthoughts_datadir'] = "/storage/ostrava1/home/nadvorj1/skip-thoughts/"
+
+    # disable GloVe
+    c['embdim'] = None
+    # disable Keras training
+    c['ptscorer'] = None
+
+    # Which version of precomputed ST vectors to use
+    c["skipthoughts_uni_bi"] = "combined"
+
+    # loss is set in __init__
+    c["loss"] = None
+
+    # Values from original code (ryankiros/skip-thoughts/eval_sick.py):
+    c['merge_sum'] = True
+    c['merge_mul'] = False
+    c['merge_diff'] = False
+    c['merge_absdiff'] = True
+    # l2=0 is used in eval_sick.py. They used some value in paper
+    c['l2reg'] = 0.0
+    c['dropout'] = 0.0
+
+    # Add <End-Of-Sentence> mark to inputs. If inputs have correct
+    # punctuation it tend to be better without EOS.
+    c['use_eos'] = True
+
+    # appending boolean flags to ST vectors
+    c["use_flags"] = False
+
+
+class STModel:
+    """ Quacks (a little) like a Keras model. """
+
+    def __init__(self, c, output):
+        self.weights_to_load = None
+        self.c = c
+        self.output = output
+
+        if c.get("clipnorm"):
+            c["opt"] = Adam(clipnorm=c["clipnorm"])
```

Comment:
It'd be great to have all tunables in the config() method so that we can easily see what can we tweak. ```c['clipnorm'] = None``` there would do the trick.